### PR TITLE
feat: [CDS-35541]: add yaml validation for serverless artifact fields - artifactPath, artifactPathFilter and artifactDirectory

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
##### Summary:
- Display toaster error and stop saving process if no values given for any of these fields - artifactPath, artifactPathFilter and artifactDirectory

##### Jira Links:
https://harness.atlassian.net/browse/CDS-35541

##### Screenshots:
Screenshot 2022-04-05 at 4 24 15 PM

Screenshot 2022-04-05 at 4 25 20 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 1]
```diff
@@ -213,3 +225,112 @@ export const getSelectedDeploymentType = (
   }
   return get(stage, 'stage.spec.serviceConfig.serviceDefinition.type', null)
 }
+
+const isArtifactFieldPresent = (stage: DeploymentStageElementConfigWrapper, fieldName: string): boolean => {
+  return (
+    stage.stage?.spec?.serviceConfig.serviceDefinition?.spec.artifacts?.primary?.spec[fieldName] &&
+    stage.stage?.spec?.serviceConfig.serviceDefinition?.spec.artifacts?.primary?.spec[fieldName].toString().trim()
```
[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/utils/stageHelpers.ts]: Done.

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 2]
```diff
@@ -213,3 +225,112 @@ export const getSelectedDeploymentType = (
...
+      .length > 0
+  )
+}
+
+const isArtifactFieldPresentInPropagatedStage = (
+  stage: DeploymentStageElementConfigWrapper,
+  fieldName: string
+): boolean => {
+  return (
+    stage.stage?.spec?.serviceConfig.stageOverrides?.artifacts?.primary?.spec[fieldName] &&
```
[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/utils/stageHelpers.ts]: Done.

**Comment by bhavya-sinha** [comment]
There are 9 code smells.
Few like unnecessary assertions can be fixed now, except the ones for reducing CC

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
> There are 9 code smells. Few like unnecessary assertions can be fixed now, except the ones for reducing CC. @chetanvekariya-harness

@bhavya-sinha - That is happening because Satyam's changes. I will be fixing it before we merge all changes to develop.

**Pull request merged**